<a href="https://colab.research.google.com/github/IvanBovsunovskiy/math_probability/blob/hw5/HW5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import numpy as np

##__Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование__
1. Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

Решать через тест гипотезы

3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

4. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

__1. Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.__


Так как $\sigma = 16$ и надежность - 0.95, то 
$Z_{1,2} = \pm1,96$

доверительный интервал: $\left[M - Z_1\cdot\frac{\sigma}{\sqrt{n}}, M - Z_1\cdot\frac{\sigma}{\sqrt{n}}\right] \rightarrow \left[80 - 1.96\cdot\frac{16}{\sqrt{256}}, 80 + 1.96\cdot\frac{16}{\sqrt{256}}\right] \rightarrow \left[78.04, 81.96\right]$


__2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.__

In [14]:
X = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
M = np.mean(X)
sigma = np.std(X)
print(f'   M = {round(M,3)}\n std = {round(sigma,3)}')
print(M - sigma/10**0.5*1.96, M + sigma/10**0.5*1.96)


   M = 6.59
 std = 0.428
6.324928568117951 6.855071431882051


Аналогично заданию 1 имеем:
$\left[M - Z_1\cdot\frac{\sigma}{\sqrt{n}}, M - Z_1\cdot\frac{\sigma}{\sqrt{n}}\right] \rightarrow \left[6.59 - 1.96\cdot\frac{0.428}{\sqrt{10}}, 6.59 + 1.96\cdot\frac{0.428}{\sqrt{10}}\right] \rightarrow \left[6.325, 6.855\right]$


__3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.__

$M = 17, ~ \alpha = 0.05, ~ n = 100, \overline{X} = 17.5, ~ \sigma^2 = 4$
Для одностороннего критерия и так как известна дисперсия имеем $Z_{\alpha} = 1.645$
Пусть 0-вая гипотеза: 17мм - средний диаметр подшипников, $H_1$ гипотеза: средний диаметр больше 17мм.
Тогда $Z_H = \frac{\overline{X} - M}{\sigma/\sqrt{n}} = \frac{17.5 - 17}{2/\sqrt{100}} = 2.5 > Z_{\alpha} \Rightarrow$ 0-вая гипотеза неверна и средний диаметр __не__ равен 17мм

__4. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?__

In [18]:
X = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]
M = np.mean(X)
sigma = np.std(X)
print(f'   M = {round(M,3)}\n std = {round(sigma,3)}')
(198.5-200)/(sigma/10**0.5)

   M = 198.5
 std = 4.225


-1.1227217828476797

$M = 200, ~ \alpha = 0.01, ~ n = 10, \overline{X} = 198.5, ~ \sigma = 4.225$
Для одностороннего критерия и так как известна дисперсия имеем $Z_{\alpha} = - 2.3233$
Пусть 0-вая гипотеза: 200г - средний вес пачки, $H_1$ гипотеза: средний вес меньше 200г.
Тогда $Z_H = \frac{\overline{X} - M}{\sigma/\sqrt{n}} = \frac{198.5-200}{4.225/\sqrt{10}} = -1.123 > Z_{\alpha} \Rightarrow$ 0-вая гипотеза верна и средний вес __равен__ 200г